lib/ansible/modules/cloud/smartos/nictagadm.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Bruce Smith <Bruce.Smith.IT@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nictagadm
+short_description: Manage NIC tags on SmartOS systems.
+description:
+  - Create of delete NIC tags on SmartOS systems.
+version_added: "2.5"
+author: Bruce Smith (@SmithX10)
+options:
+    name:
+        description:
+            - NIC tag name.
+        required: true
+    mac:
+        description:
+            - MAC Address to attach the NIC tag to when not using an etherstub. mac and etherstub are mutually exclusive.
+        required: false
+        default: None
+    etherstub:
+        description:
+            - Specifies that the NIC tag will be attached to a created etherstub. etherstub is mutually exclusive with both mtu, and mac.
+        required: false
+        default: false
+    mtu:
+        description:
+            - MTU size of the NIC tag. mtu and etherstub are mutually exclusive.
+        required: false
+        default: None
+    force:
+        description:
+            - When State.Absent is set this switch will use the -f parameter and delete the NIC tag regardless of existing VMs
```

Comment:
Would be better formatted as "When I(state) is absent this switch will use the C(-f) [...]" and there's a missing full stop.